### DISCUSSION TITLE
Are there plans to bring back metrictest?

### DISCUSSION THREAD

**Discussion by mfridman** [open-telemetry/opentelemetry-go#5437] [General]
There used to be a [`metrictest` package](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/metrictest#NewTestMeterProvider) that made it possible to test metrics collection in testing code.

I see it has been removed, but are there plans to bring it back? 

If not, is there a recommendation on testing a `metric.MeterProvider`. Is there _something_ in the Go otel package that could satisfy that interface?

**Comment by dmathieu** [comment]
There are assertions available in the [metricdatatest](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric@v1.27.0/metricdata/metricdatatest) package.

The idea is to use this with a manual reader in your tests:

```golang
rdr := sdkmetric.NewManualReader()
mp := sdkmetric.NewMeterProvider(sdkmetric.WithReader(rdr))
```

Pass the meter provider so it's used within your tested code.
And when you want to retrieve metrics data (to use the provided assertions, or write your own), you can use the reader's [Collect](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#NewManualReader) method.

```golang
var data metricdata.ResourceMetrics
err := rdr.Collect(context.Background(), &data)

// Check for no error, and that the provided data is as expected.
```
